src/gles_fns.rs

```diff
@@ -0,0 +1,1311 @@
+// Copyright 2014 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+pub struct GlesFns {
+    ffi_gl_: GlesFfi,
+}
+
+impl GlesFns
+{
+    fn new(ffi_gl_: GlesFfi) -> Arc<GlesFns> {
```

Comment:
Arc was used to simplify the gl object usage management and to handle delivering a reference of gl::Gl to background thread.

If Arc is used, it becomes easier to replace static gl function calls by gl::Gl object. I already confirmed it in bug 1325911 as wip patch.
> https://bugzilla.mozilla.org/show_bug.cgi?id=1325911